paramiko/proxy.py

```diff
@@ -119,7 +119,10 @@ def recv(self, size):
             raise ProxyCommandFailure(" ".join(self.cmd), e.strerror)
 
     def close(self):
-        os.kill(self.process.pid, signal.SIGTERM)
+        # The subprocess may be killed in other ways (For example, asynchronous)
+        # Before killing this subprocess, you need to determine whether it exists
+        if not self.closed:
+            os.kill(self.process.pid, signal.SIGTERM)
```

Comment:
We could use [`Popen.terminate()`](https://docs.python.org/3/library/subprocess.html#subprocess.Popen.terminate) simply, in order to terminate the process if alive.

See
* posix - https://github.com/python/cpython/blob/v3.6.15/Lib/subprocess.py#L1596-L1605
* nt - https://github.com/python/cpython/blob/v3.6.15/Lib/subprocess.py#L1153-L1166

```suggest
        self.process.terminate()
```

Reply:
Thank you for your guidance.

Reply:
@nuonuonn Thanks for the explain.

The issue has been fixed in Python 3.10.0, https://github.com/python/cpython/commit/01a202ab6b0ded546e47073db6498262086c52e9

However, the issue occurs for the version or early. We could change like the following:

```suggest
diff --git a/paramiko/proxy.py b/paramiko/proxy.py
index f7609c984..111e91627 100644
--- a/paramiko/proxy.py
+++ b/paramiko/proxy.py
@@ -119,7 +119,13 @@ class ProxyCommand(ClosingContextManager):
             raise ProxyCommandFailure(" ".join(self.cmd), e.strerror)

     def close(self):
-        os.kill(self.process.pid, signal.SIGTERM)
+        try:
+            self.process.terminate()
+        except ProcessLookupError:
+            pass  # Suppress the error if the process is not existent
+        # Close pipes and avoid zombie process
+        with self.process:
+            pass

     @property
     def closed(self):
```